### PR TITLE
Vite 5

### PR DESCRIPTION
**Pull request by quentinderoubaix**

### CODE REVIEW

File: demo/scripts/emitFile.plugin.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import type {Plugin} from 'vite';
-import type {EmittedFile} from 'rollup';
+import type {EmittedFile} from 'vite/node_modules/rollup';
```
Comment (divdavem) [inline demo/scripts/emitFile.plugin.ts:2]: Why is this necessary?
`node_modules` should never appear in imports.

**Comment by divdavem** [comment]
It looks like [Vite 5 is not officially supported by SvelteKit yet](https://github.com/sveltejs/kit/issues/11062).
If, from this PR, I remove the `package-lock.json` file and the `node_modules` folders and I run `npm install`, I am getting an error when resolving `vite` because version 5 is not compatible with `@sveltejs/kit`:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @agnos-ui/demo@undefined
npm ERR! Found: vite@4.5.0
npm ERR! demo/node_modules/vite
npm ERR!   peer vite@"^4.0.0" from @sveltejs/kit@1.27.6
npm ERR!   demo/node_modules/@sveltejs/kit
npm ERR!     dev @sveltejs/kit@"^1.27.6" from @agnos-ui/demo@undefined
npm ERR!     demo
npm ERR!       @agnos-ui/demo@undefined
npm ERR!       node_modules/@agnos-ui/demo
npm ERR!         workspace demo from the root project
npm ERR!     peer @sveltejs/kit@"^1.5.0" from @sveltejs/adapter-static@2.0.3
npm ERR!     demo/node_modules/@sveltejs/adapter-static
npm ERR!       dev @sveltejs/adapter-static@"^2.0.3" from @agnos-ui/demo@undefined
npm ERR!       demo
npm ERR!         @agnos-ui/demo@undefined
npm ERR!         node_modules/@agnos-ui/demo
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! dev @sveltejs/vite-plugin-svelte@"^3.0.1" from @agnos-ui/demo@undefined
npm ERR! demo
npm ERR!   @agnos-ui/demo@undefined
npm ERR!   node_modules/@agnos-ui/demo
npm ERR!     workspace demo from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: vite@5.0.2
npm ERR! node_modules/vite
npm ERR!   peer vite@"^5.0.0" from @sveltejs/vite-plugin-svelte@3.0.1
npm ERR!   node_modules/@sveltejs/vite-plugin-svelte
npm ERR!     dev @sveltejs/vite-plugin-svelte@"^3.0.1" from @agnos-ui/demo@undefined
npm ERR!     demo
npm ERR!       @agnos-ui/demo@undefined
npm ERR!       node_modules/@agnos-ui/demo
npm ERR!         workspace demo from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Have you run with `npm install` with `--force` to update `package-lock.json` with incompatible versions of `sveltejs/kit` and `vite`?

**Comment by quentinderoubaix** [comment]
> It looks like [Vite 5 is not officially supported by SvelteKit yet](https://github.com/sveltejs/kit/issues/11062). If, from this PR, I remove the `package-lock.json` file and the `node_modules` folders and I run `npm install`, I am getting an error when resolving `vite` because version 5 is not compatible with `@sveltejs/kit`:
> 
> ```
> npm ERR! code ERESOLVE
> npm ERR! ERESOLVE could not resolve
> npm ERR! 
> npm ERR! While resolving: @agnos-ui/demo@undefined
> npm ERR! Found: vite@4.5.0
> npm ERR! demo/node_modules/vite
> npm ERR!   peer vite@"^4.0.0" from @sveltejs/kit@1.27.6
> npm ERR!   demo/node_modules/@sveltejs/kit
> npm ERR!     dev @sveltejs/kit@"^1.27.6" from @agnos-ui/demo@undefined
> npm ERR!     demo
> npm ERR!       @agnos-ui/demo@undefined
> npm ERR!       node_modules/@agnos-ui/demo
> npm ERR!         workspace demo from the root project
> npm ERR!     peer @sveltejs/kit@"^1.5.0" from @sveltejs/adapter-static@2.0.3
> npm ERR!     demo/node_modules/@sveltejs/adapter-static
> npm ERR!       dev @sveltejs/adapter-static@"^2.0.3" from @agnos-ui/demo@undefined
> npm ERR!       demo
> npm ERR!         @agnos-ui/demo@undefined
> npm ERR!         node_modules/@agnos-ui/demo
> npm ERR! 
> npm ERR! Could not resolve dependency:
> npm ERR! dev @sveltejs/vite-plugin-svelte@"^3.0.1" from @agnos-ui/demo@undefined
> npm ERR! demo
> npm ERR!   @agnos-ui/demo@undefined
> npm ERR!   node_modules/@agnos-ui/demo
> npm ERR!     workspace demo from the root project
> npm ERR! 
> npm ERR! Conflicting peer dependency: vite@5.0.2
> npm ERR! node_modules/vite
> npm ERR!   peer vite@"^5.0.0" from @sveltejs/vite-plugin-svelte@3.0.1
> npm ERR!   node_modules/@sveltejs/vite-plugin-svelte
> npm ERR!     dev @sveltejs/vite-plugin-svelte@"^3.0.1" from @agnos-ui/demo@undefined
> npm ERR!     demo
> npm ERR!       @agnos-ui/demo@undefined
> npm ERR!       node_modules/@agnos-ui/demo
> npm ERR!         workspace demo from the root project
> npm ERR! 
> npm ERR! Fix the upstream dependency conflict, or retry
> npm ERR! this command with --force or --legacy-peer-deps
> npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
> ```
> 
> Have you run `npm install` with `--force` to update `package-lock.json` with incompatible versions of `@sveltejs/kit` and `vite`?

I not update the dependencies using --force or legacy-peer-deps, so quite surprised to see this.
Thanks for checking, will put that PR as Draft until svelte kit officially supports vite 5.

**Comment by divdavem** [comment]
@quentinderoubaix I have opened PR #330 to update both SvelteKit and Vite. Do you think we can close this one?
